src/server/cmd/pachctl/cmd/cmd_test.go

```diff
@@ -94,3 +95,41 @@ func testDeploy(t *testing.T, devFlag bool, noMetrics bool, expectedEnvValue boo
 	}
 	require.Equal(t, true, foundPachdManifest)
 }
+
+func testBadAWSCredentialsDeploy(t *testing.T) {
+	// Setup user config prior to test
+	// So that stdout only contains JSON no warnings
+	_, err := config.Read()
+	require.NoError(t, err)
+	noMetrics := false
+
+	// pachctl deploy amazon ${BUCKET_NAME} "${AWS_ID}" "${AWS_KEY}" " " ${AWS_REGION} ${STORAGE_NAME} ${STORAGE_SIZE}
+	os.Args = []string{
+		"deploy",
+		"amazon",
+		"--dry-run",
+		"abucket",
+		"\"AKIASDLKJG346LK99JGA\"",                     // Not a real id
+		"\"2rDFgllkj445LK/SDFLjj345lkj57+4564LKsf82\"", // See that slash! It's bad!
+		"\" \"",
+		"us-west2",
+		"vol-2345436",
+		"100",
+	}
+	// This should exit 1
+	err = deploycmds.DeployCmd(&noMetrics).Execute()
+}
+
+func TestBadAWSCredentialsDeploy(t *testing.T) {
+	if os.Getenv("BE_CRASHER") == "1" {
+		testBadAWSCredentialsDeploy(t)
+		return
+	}
+	cmd := exec.Command(os.Args[0], "-test.run=TestBadAWSCredentialsDeploy")
+	cmd.Env = append(os.Environ(), "BE_CRASHER=1")
```

Comment:
This is insane. Your shelling this test out to run itself again (which will cause the test to be recompiled) to check if a function returns an error when it's passed a string with a `/` in it.